src/lib/analytics/__tests__/segmentOneTrustIntegration.jest.ts

```diff
@@ -0,0 +1,315 @@
+      window.location.reload = jest.fn()
+      window.analytics = {
+        initialized: true,
+        load: jest.fn(),
+      } as any
+    })
+    it("reloads page if shouldReload is true, and exit", () => {
+      SOI.conditionallyLoadAnalytics({
+        writeKey: "abc",
+        destinationPref: {},
+      })
+      expect(window.location.reload).toHaveBeenCalled()
+      // @ts-ignore
+      expect(window.analytics.load).not.toHaveBeenCalled()
+    })
+    it("does not reload page if shouldReload is false, and exit", () => {
+      SOI.conditionallyLoadAnalytics({
+        writeKey: "abc",
+        destinationPref: {},
+        shouldReload: false,
+      })
+      expect(window.location.reload).not.toHaveBeenCalled()
+      // @ts-ignore
+      expect(window.analytics.load).not.toHaveBeenCalled()
+    })
+  })
+  describe("segment not yet initialized", () => {
+    beforeEach(() => {
+      window.location.reload = jest.fn()
+      window.analytics = {
+        initialized: false,
+        load: jest.fn(),
+      } as any
+    })
+    it("loads analytics if any destination is enabled", () => {
+      SOI.conditionallyLoadAnalytics({
+        writeKey: "abc",
+        destinationPref: { foo: true, bar: false },
+      })
+      expect(window.location.reload).not.toHaveBeenCalled()
+      // @ts-ignore
+      expect(window.analytics.load).toHaveBeenCalled()
+    })
+    it("does not load analytics if no destination is enabled", () => {
+      SOI.conditionallyLoadAnalytics({
+        writeKey: "abc",
+        destinationPref: { foo: false },
+      })
+      expect(window.location.reload).not.toHaveBeenCalled()
+      // @ts-ignore
+      expect(window.analytics.load).not.toHaveBeenCalled()
+    })
+    it("does not load analytics if destinationPref is empty", () => {
+      SOI.conditionallyLoadAnalytics({
+        writeKey: "abc",
+        destinationPref: {},
+      })
+      expect(window.location.reload).not.toHaveBeenCalled()
+      // @ts-ignore
+      expect(window.analytics.load).not.toHaveBeenCalled()
+    })
+  })
+})
+
+describe("validateSegmentResponse", () => {
+  it("returns true if destinations is an empty array", () => {
+    expect(SOI.validateSegmentResponse([])).toBe(true)
+  })
+  it("returns true if destinations has good data", () => {
+    expect(SOI.validateSegmentResponse([{ creationName: "foo" }])).toBe(true)
+  })
+  it("returns false if destinations is not an array", () => {
+    expect(SOI.validateSegmentResponse("foo")).toBe(false)
+  })
+  it("returns false if a destination has no creationName key", () => {
+    const destinations = [{ creationName: "foo" }, { blah: "blah" }]
+    expect(SOI.validateSegmentResponse(destinations)).toBe(false)
+  })
+  it("returns false if a destination has creationName key but its value is not a string", () => {
+    const destinations = [{ creationName: "foo" }, { blah: {} }]
+    expect(SOI.validateSegmentResponse(destinations)).toBe(false)
+  })
+})
```

Comment:
@artsyjian - i think the testing approach you've laid out here is totally fine 👍